code/edgeidentity/src/test/java/com/adobe/marketing/mobile/edge/identity/IdentityStateTests.java

```diff
@@ -542,4 +562,296 @@ public void testUpdateLegacyExperienceCloudId_notSetWhenExistingIsNull() {
+	//     assertEquals state.getIdentityProperties().getAdId()
+	// 4. Identity props saved to persistence
+	//     mockSharedPreferenceEditor
+	//
+	// Expected values should be hard-coded to prevent tautological test (one where the test must pass by definition,
+	// because the expectation uses the same logic as the code being tested); has effect of checking:
+	// - Exact text - values and case sensitivity
+	// - Hierarchy of event data object
+
+	private void assertUpdateAdvertisingIdentifier_updatedWithConsentChange(
+		String persistedAdId,
+		String newAdId,
+		String expectedAdId,
+		String expectedConsent
+	) throws Exception {
+		// Setup
+		IdentityState state = new IdentityState((new IdentityProperties()));
+		state.getIdentityProperties().setECID(new ECID());
+		state.getIdentityProperties().setAdId(persistedAdId);
+
+		Event event = fakeGenericIdentityEvent(newAdId);
+		state.updateAdvertisingIdentifier(event, mockSharedStateCallback);
+
+		// Verify consent event
+		if (expectedConsent == null) {
+			PowerMockito.verifyStatic(MobileCore.class, Mockito.never());
+			MobileCore.dispatchEvent(any(Event.class), any(ExtensionErrorCallback.class));
+		} else {
+			final ArgumentCaptor<Event> consentEventCaptor = ArgumentCaptor.forClass(Event.class);
+			PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
+			MobileCore.dispatchEvent(consentEventCaptor.capture(), any(ExtensionErrorCallback.class));
+
+			Event consentEvent = consentEventCaptor.getAllValues().get(0);
+
+			Map<String, String> consentEventData = flattenMap(consentEvent.getEventData());
+			System.out.println(consentEventData);
+			// `flattenMap` allows for checking the keys' hierarchy and literal values simultaneously
+			assertEquals("GAID", consentEventData.get("consents.adID.idType"));
+			assertEquals(expectedConsent, consentEventData.get("consents.adID.val"));
+		}
+
+		// Verify persistent store
+		final ArgumentCaptor<String> persistenceValueCaptor = ArgumentCaptor.forClass(String.class);
+		verify(mockSharedPreferenceEditor, times(1))
+			.putString(eq(IdentityConstants.DataStoreKey.IDENTITY_PROPERTIES), persistenceValueCaptor.capture());
+		Map<String, String> persistedData = flattenJSONString(persistenceValueCaptor.getAllValues().get(0));
+
+		if (expectedAdId != null) {
+			assertEquals(6, persistedData.size()); // updated ad ID + ECID
+			assertEquals("false", persistedData.get("identityMap.GAID[0].primary"));
+			assertEquals(expectedAdId, persistedData.get("identityMap.GAID[0].id"));
+			assertEquals("ambiguous", persistedData.get("identityMap.GAID[0].authenticatedState"));
+		} else {
+			assertEquals(3, persistedData.size()); // ECID
+		}
+		assertEquals("false", persistedData.get("identityMap.ECID[0].primary"));
+		assertEquals(state.getIdentityProperties().getECID().toString(), persistedData.get("identityMap.ECID[0].id"));
+		assertEquals("ambiguous", persistedData.get("identityMap.ECID[0].authenticatedState"));
```

Comment:
It looks like you perform this check several times. Can you extract this out to a common function?

Reply:
Extracted!